src/Product/AttributeMapping/AttributeMappingHelper.php

```diff
@@ -0,0 +1,316 @@
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\IsBundle;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Material;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\MPN;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Multipack;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Pattern;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Size;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\SizeSystem;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\SizeType;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\WithMappingInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Helper Class for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product\AttributeMapping
+ */
+class AttributeMappingHelper implements Service {
+
+
+	private const ATTRIBUTES_AVAILABLE_FOR_MAPPING = [
+		Adult::class,
+		AgeGroup::class,
+		Brand::class,
+		Color::class,
+		Condition::class,
+		Gender::class,
+		GTIN::class,
+		IsBundle::class,
+		Material::class,
+		MPN::class,
+		Multipack::class,
+		Pattern::class,
+		Size::class,
+		SizeSystem::class,
+		SizeType::class,
+	];
+
+	public const CATEGORY_CONDITION_TYPE_ALL    = 'ALL';
+	public const CATEGORY_CONDITION_TYPE_ONLY   = 'ONLY';
+	public const CATEGORY_CONDITION_TYPE_EXCEPT = 'EXCEPT';
+
+	/**
+	 * @var Transients|null
+	 */
+	private static ?Transients $transients = null;
+
+	/**
+	 * Gets all the available attributes for mapping
+	 *
+	 * @return array
+	 */
+	public function get_attributes(): array {
+		$destinations = [];
+
+		/**
+		 * @var WithMappingInterface $attribute
+		 */
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			array_push(
+				$destinations,
+				[
+					'id'    => $attribute::get_id(),
+					'label' => $attribute::get_name(),
+					'enum'  => $attribute::is_enum(),
+				]
+			);
+		}
+
+		return $destinations;
+	}
+
+	/**
+	 * Gets all the available sources identified by attribute key
+	 *
+	 * @return array
+	 */
+	public function get_sources(): array {
+		$sources = [];
+
+		/**
+		 * @var AttributeInterface $attribute
+		 */
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+
+			$attribute_sources = [];
+
+			foreach ( $attribute::get_sources() as $key => $value ) {
+				array_push(
+					$attribute_sources,
+					[
+						'id'    => $key,
+						'label' => $value,
+					]
+				);
+			}
+
+			$sources[ $attribute::get_id() ] = $attribute_sources;
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * Returns a single instance of Transients
+	 *
+	 * @return Transients A single instance of Transients
+	 */
+	public static function get_transients_object(): Transients {
+		if ( is_null( self::$transients ) ) {
+			self::$transients = new Transients();
+		}
+
+		return self::$transients;
+	}
```

Comment:
The meta query is called per each attribute under Product/Attributes folder which as far as I know has not access to container.

Reply:
What I mean is that even if we pass the Transient object in the constructor and the Provider. It would be able to be called inside a static function? 🤔

Reply:
Ah ok I see what you mean. I missed that the attributes were all static, another reason to move the `get_fields` function to the trait to reduce confusion. Anyways, if we are removing the Transient for now we don't need to worry about this yet.